src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +39,116 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCurrentDir($dir)
+    {
+        $this->currentDir = $dir;
+        parent::setCurrentDir($dir);
+    }
+
+    public function registerClasses(Definition $prototype, $psr4Glob, $directory)
```

Comment:
Should be `protected`, right?
I also suggest to make it `final` as I don't think it is useful to extend it.